File: server/model/common/token.go
package common

type Token struct {
	AccessToken       string `json:"access_token"`        // 访问令牌
	ExpireTime        int64  `json:"expire_time"`         // 访问令牌过期时间
	RefreshToken      string `json:"refresh_token"`       // 刷新令牌
	RefreshExpireTime int64  `json:"refresh_expire_time"` // 刷新令牌过期时间
	// MustChangePassword 标记该账号仍需强制修改密码，前端据此强制跳转改密页
	MustChangePassword bool `json:"must_change_password"`
	// TenantId 当前 token 所属租户ID，前端据此定位当前租户（名称经 my_tenants 查询）
	TenantId string `json:"tenant_id"`
	// IsSuperAdmin 平台超级管理员标识。仅平台超管登录时返回 true（omitempty 使
	// 普通账号响应中不含该字段），前端据此区分平台域与业务租户。
	IsSuperAdmin bool `json:"is_super_admin,omitempty"`
}

File: web/src/lib/token.ts
import { addSeconds } from 'date-fns'
import type { LoginResult, Token } from '~/do-exercise-api'

// applyToken 将登录/切换租户返回的 LoginResult 归一化为 Token 并写入前端缓存结构：
// 换算绝对过期时间，并给可选字段补默认值（登录成功路径必然携带 token 字段）。
export function applyToken(token: LoginResult): Token {
  return {
    ...token,
    accessToken: token.accessToken ?? '',
    refreshToken: token.refreshToken ?? '',
    mustChangePassword: token.mustChangePassword ?? false,
    expireTime: addSeconds(new Date(), token.expireTime ?? 0).getTime(),
    refreshExpireTime: addSeconds(
      new Date(),
      token.refreshExpireTime ?? 0
    ).getTime(),
  } as Token
}